src/entity/entity.cpp

```diff
@@ -320,70 +322,127 @@ void EntitySystem::send_initial_state(ClientHandle *handle) {
     }
 }
 
-void EntitySystem::draw() {
+
 #ifdef IMGUI_ENABLE
+using ImGuiDisplayFunc = std::function<void(const char *, void *)>;
+static std::unordered_map<std::size_t, ImGuiDisplayFunc> func_map;
+
+namespace ImGuiFuncs {
+    void empty_f(const char *name, void *) {}
+
+    void bool_t(const char *name, void * v) {
+        ImGui::Checkbox(name, (bool *) v);
+    }
+
+    void vec3_t(const char *name, void * v) {
+        ImGui::ColorEdit3(name, (f32 *) v);
+    }
+}
```

Comment:
Later we'd want this in another file, I think

Reply:
Yeah, I think about adding it in another PR, where I just add a bunch of these functions. And refactor it to another file. I'll make an issue.